src/syntax-highlight/syntax-highlight.js

```diff
@@ -69,19 +69,36 @@ export default function syntaxHighlight(diff, afterWordDiff) {
     }
 }
 
-function syntaxHighlightSourceCodeLines($diff) {
-    const sourceLines = [
-        ...$diff.find('pre:not([class*=language]), pre:has(ins), pre:has(del)'),
-    ]
-
-    sourceLines.forEach(preElement => {
-        if (!preElement.firstChild.$$rbb_isSyntaxHighlighted) {
-            Prism.highlightElement(preElement)
-            preElement.classList.add('__rbb_syntax_highlight')
-            // eslint-disable-next-line camelcase
-            preElement.firstChild.$$rbb_isSyntaxHighlighted = true
-        }
-    })
+async function syntaxHighlightSourceCodeLines($diff, querySelector) {
+    const sourceLines = [...$diff.find(querySelector)]
+
+    const promises = sourceLines.map(
+        preElement =>
+            new Promise((resolve, reject) => {
+                const { classList, firstChild, innerText } = preElement
+
+                if (firstChild.$$rbb_isSyntaxHighlighted) {
+                    reject('Already highlighted')
+                    return
+                }
+
+                // Lines over the arbitrary max length of 9999 will be considered as minified
+                if (innerText && innerText.length > 9999) {
+                    reject(`Line is too long, probably minified`)
+                    return
+                }
+
+                Prism.highlightElement(preElement)
```

Comment:
Maybe we can use web worker, but Prism warn about using async highlight... web workers are slower and just limiting the text to a max length of 9999 is quite enough. Sure it will skip this line, but the ui does not freezes anymore

```jsx
Prims.highlightElement(element, true, () => {}); // async with web worker
```